backend/app/model/mixins/relationships.rb

```diff
@@ -92,58 +92,123 @@ def self.transfer(target, victims)
 
     victims_by_model = victims.reject {|v| (v.class == target.class) && (v.id == target.id)}.group_by(&:class)
 
-    victims_by_model.each do |victim_model, victims|
-
-      # If we're merging a record of type A with relationship R into a record of
-      # type B, type B must also support that relationship type.  If it doesn't,
-      # we risk losing data through the merge and should abort.
-      #
-      unless participating_models.include?(victim_model)
-        found = self.find_by_participant_ids(victim_model, victims.map(&:id))
-
-        unless found.empty?
-          raise ReferenceError.new("#{victim_model} to be merged has data for relationship #{self}, but target record doesn't support it.")
-        end
-      end
+    victims_by_model.each do |victim_model, vics|
 
+      confirm_accepts_target(victim_model, vics)
       victim_columns = self.reference_columns_for(victim_model)
 
       victim_columns.each do |victim_col|
 
-        # Find any relationship where the current column contains a reference to
-        # our victim
-        self.filter(victim_col => victims.map(&:id)).each do |relationship|
+        vics.each do |victim|
+
+          parents = who_participates_with(victim)
+          next if parents.empty?
```

Comment:
When return value is an array can just use directly: `who_participates_with(victim).each do |parent|`